lib/Reader/abisignature.cpp

```diff
@@ -338,6 +354,12 @@ Value *ABICallSignature::emitCall(GenIR &Reader, Value *Target, bool MayThrow,
         Reader.getUnmanagedPointerType(Result.getType());
     Arguments[ResultIndex] = ResultNode =
         Reader.createTemporary(Result.getType());
+
+    Type *ResultTy = ResultNode->getType();
+    if (ResultTy->isPointerTy() &&
```

Comment:
I would rewrite this check as `if (Result.getType()->isStructTy())`; the `isPointerTy()` check is redundant.